mesonbuild/compilers.py

```diff
@@ -2729,3 +2733,135 @@ def unix_compile_flags_to_native(self, args):
 
     def get_link_debugfile_args(self, targetfile):
         return []
+
+
+# Only tested on linux for ICC 17.0.0
+#    * object extraction does not compile (don't know) (25 object extraction fails)
```

Comment:
Where should the spaces be  escaped?
Overloading these functions und replace any occurrence of ' ' by '\ '?

`````
   def get_dependency_gen_args(self, outtarget, outfile):
        return ['-MMD', '-MQ', outtarget, '-MF', outfile]

    def depfile_for_object(self, objfile):
        return objfile + '.' + self.get_depfile_suffix()
````